File: intro/digitDegree.py
# https://app.codesignal.com/arcade/intro/level-9/hoLtYWbjdrD2PF6yo

def digitDegree(n):
    if n < 10:
        return 0

    sumOfDigits = sum([int(i) for i in str(n)])
    return digitDegree(sumOfDigits) + 1


def digitDegree(n):
    d = 0
    while n >= 10:
        n = sum([int(i) for i in str(n)])
        d += 1
    return d

print(digitDegree(10))

File: intro/digitDegree.js
// https://app.codesignal.com/arcade/intro/level-9/hoLtYWbjdrD2PF6yo

function digitDegree(n) {
    let arrNumber = n.toString().split('').map(x => parseInt(x, 10));
    let count = 0;
    if (n < 10) {
        return count;
    }
    while (true) {
        let sum = arrNumber.reduce((a, b) => a + b, 0)
        count += 1
        if (sum < 10) {
            return count
        } else {
            arrNumber = sum.toString().split('').map(x => parseInt(x, 10));
        }

    }
}